Offseason2022/src/main/java/frc/robot/subsystems/TowerConveyor.java

```diff
@@ -38,6 +90,71 @@ public void simulationPeriodic( )
     // This method will be called once per scheduler run when in simulation
   }
 
+  public void initialize( )
+  {
+    DataLogManager.log("VC Init");
+    setVerticalConveyorSpeed(Mode.VCONVEYOR_STOP);
+  }
+
+  public void faultDump( )
+  {
+    PhoenixUtil.getInstance( ).talonFXFaultDump(m_motorVC9, "VC 9");
+  }
+
+  // Set mode of conveyor
+  public void setVerticalConveyorSpeed(Mode mode)
+  {
+    final String strName;
+    double outputVC = 0.0; // Default: off
+
+    switch (mode)
+    {
+      default :
+      case VCONVEYOR_STOP :
+        strName = "STOP";
+        outputVC = 0.0;
+        break;
+      case VCONVEYOR_ACQUIRE :
+        strName = "ACQUIRE";
+        outputVC = m_acquireSpeed;
+        break;
+      case VCONVEYOR_ACQUIRE_SLOW :
+        strName = "ACQUIRE_SLOW";
+        outputVC = m_acquireSpeedSlow;
+        break;
+      case VCONVEYOR_EXPEL :
+        strName = "EXPEL";
+        outputVC = m_expelSpeed;
+        break;
+      case VCONVEYOR_EXPEL_FAST :
+        strName = "EXPEL_FAST";
+        outputVC = m_expelSpeedFast;
+        break;
+    }
+
+    DataLogManager.log(getSubsystem( ) + ": VC Set Speed - " + strName);
+
+    if (m_talonValidVC9)
+      m_motorVC9.set(ControlMode.PercentOutput, outputVC);
+  }
+
+  boolean isCargoDetected( )
+  {
+    boolean cargoDetected;
+    if (m_cargoDetected.get( ) == 0)
+    {
+      cargoDetected = true;
+
+    }
+    else
+    {
+      cargoDetected = false;
+    }
+
+    SmartDashboard.putBoolean("VC_cargoDetected", cargoDetected);
+    return cargoDetected;
+  }
+
   // Put methods for controlling this subsystem

```

Comment:
Our new methods should go below "// Put methods for..." as a general style guideline.